templates/generate-readme.py

```diff
@@ -0,0 +1,156 @@
+  artifact_id: str = None
+
+  def __init__(self, repo: dict):
+    self.repo = repo['repo']
+    # For now, strip out "Google Cloud" to standardize the titles
+    self.title = repo['name_pretty'].replace("Google ", "").replace("Cloud ", "")
+    self.release_level = repo['release_level']
+    artifact_parts = repo['distribution_name'].split(':')
+    if len(artifact_parts) > 1:
+      self.group_id = artifact_parts[0]
+      self.artifact_id = artifact_parts[1]
+    else:
+      self.artifact_id = repo['distribution_name']
+
+  # For sorting, we want to sort by release level, then API pretty_name
+  def __lt__(self, other):
+    if self.release_level == other.release_level:
+      return self.title < other.title
+
+    return other.release_level < self.release_level
+
+  def __repr__(self):
+    return repr((self.release_level, self.title))
+
+def replace_content_in_readme(content_rows: List[str]) -> None:
+  START_MARKER = "[//]: # (API_TABLE_START)"
+  END_MARKER = "[//]: # (API_TABLE_END)"
+  newlines = []
+  repl_open = False
+  with open("README.md", "r") as f:
+    for line in f:
+      if not repl_open:
+        newlines.append(line)
+
+      if line.startswith(START_MARKER):
+        repl_open = True
+        newlines = newlines + content_rows
+      elif line.startswith(END_MARKER):
+        newlines.append("\n")
+        newlines.append(line)
+        repl_open = False
+
+  with open("README.md", "w") as f:
+    for line in newlines:
+      f.write(line)
+
+RELEASE_LEVEL_CONTENT = {
+  "preview": "[![preview][preview-stability]][preview-description]",
+  "stable": "[![stable][stable-stability]][stable-description]"
+}
+
+def client_row(client: CloudClient) -> str:
+  maven_badge = f"[![Maven](https://img.shields.io/maven-central/v/{client.group_id}/{client.artifact_id}.svg)](https://search.maven.org/search?q=g:{client.group_id}%20AND%20a:{client.artifact_id}&core=gav)"
+  return f"| [{client.title}](https://github.com/{client.repo}) | {RELEASE_LEVEL_CONTENT[client.release_level]} | {maven_badge} |\n"
+
+def generate_table_contents(clients: List[CloudClient]) -> List[str]:
+  content_rows = [
+    "\n",
+    "| Client | Release Level | Version |\n",
+    "| ------ | ------------- | ------- |\n",
+  ]
+  return content_rows + [client_row(client) for client in clients]
+
+
+REPO_METADATA_URL_FORMAT = "https://raw.githubusercontent.com/{repo_slug}/main/.repo-metadata.json"
+
+def client_for_repo(repo_slug) -> Optional[CloudClient]:
+  url = REPO_METADATA_URL_FORMAT.format(repo_slug=repo_slug)
+  response = requests.get(url)
+  if response.status_code != requests.codes.ok:
+    return
+
+  return CloudClient(response.json())
+
+def client_for_module(module) -> Optional[CloudClient]:
+  with open ('%s/.repo-metadata.json' % module, "r") as metadata_file:
+    data = json.load(metadata_file)
+    monorepo_module = data['repo'].replace('googleapis/', '')
+    data['repo'] = 'googleapis/google-cloud-java/tree/main/%s' % monorepo_module
```

Comment:
The modules in this monorepo should reference this google-cloud-java repository.